src/components/structures/MatrixChat.js

```diff
@@ -304,6 +305,9 @@ module.exports = React.createClass({
                 });
                 break;
             case 'view_room':
+                // by default we autoPeek rooms, unless we were called explicitly with
+                // autoPeek=false by something like RoomDirectory who has already peeked
+                this.setState({ autoPeek : payload.auto_peek === false ? false : true });
```

Comment:
no, as the comment says, we set state's autoPeek to false if payload's auto_peek is false.  Otherwise we set it to true (even if payload's auto_peek was undefined).